server/tests/src/main/scala/sttp/tapir/server/tests/CreateServerTest.scala

```diff
@@ -32,9 +34,22 @@ trait CreateServerTest[F[_], +R, OPTIONS, ROUTE] {
       runTest: (SttpBackend[IO, Fs2Streams[IO] with WebSockets], Uri) => IO[Assertion]
   ): Test
 
+  def testServerLogicWithStop(
+      e: ServerEndpoint[R, F],
+      testNameSuffix: String = "",
+      interceptors: Interceptors = identity,
+      gracefulShutdownTimeout: Option[FiniteDuration] = None
+  )(
+      runTest: IO[Unit] => (SttpBackend[IO, Fs2Streams[IO] with WebSockets], Uri) => IO[Assertion]
+  ): Test = testServerLogic(e, testNameSuffix, interceptors)(runTest(IO.unit))
+
   def testServer(name: String, rs: => NonEmptyList[ROUTE])(
       runTest: (SttpBackend[IO, Fs2Streams[IO] with WebSockets], Uri) => IO[Assertion]
   ): Test
+
+  def testServerWithStop(name: String, rs: => NonEmptyList[ROUTE], gracefulShutdownTimeout: Option[FiniteDuration])(
```

Comment:
similarly here ... maybe we can reduce the number of variants here by just supporting the stop-variants, and the others would delegate here

Reply:
are these used in fact? :)

Reply:
ah ok they are ... still seems we might have to many similar variants, making this hard to understand

Reply:
Yes, I didn't want to mess up other servers, so I thought to make the default implementation implement `testServerWithStop` in terms of `testServer`. I'll add some scaladoc to make it less confusing for now.